deepchem/models/torch_models/modular.py

```diff
@@ -342,3 +271,93 @@ def fit_generator(self,
         time2 = time.time()
         logger.info("TIMING: model fitting took %0.3f s" % (time2 - time1))
         return last_avg_loss
+
+    def load_from_pretrained(  # type: ignore
+            self,
+            source_model: Optional["ModularTorchModel"] = None,
+            components: Optional[List[str]] = None,
+            checkpoint: Optional[str] = None,
+            model_dir: Optional[str] = None,
+            inputs: Optional[Sequence[Any]] = None,
+            **kwargs) -> None:
+        """Copies parameter values from a pretrained model. The pretrained model can be loaded as a source_model (ModularTorchModel object), checkpoint (pytorch .ckpt file) or a model_dir (directory with .ckpt files).
+        Specific components can be chosen by passing a list of strings with the desired component names. If both a source_model and a checkpoint/model_dir are loaded, the source_model weights will be loaded.
+
+        Parameters
+        ----------
+        source_model: dc.ModularTorchModel, required
+            source_model can either be the pretrained model or a dc.TorchModel with
+            the same architecture as the pretrained model. It is used to restore from
+            a checkpoint, if value_map is None and to create a default assignment map
+            if assignment_map is None
+        checkpoint: str, default None
+            the path to the checkpoint file to load.  If this is None, the most recent
+            checkpoint will be chosen automatically.  Call get_checkpoints() to get a
+            list of all available checkpoints
+        model_dir: str, default None
+            Restore source model from custom model directory if needed
+        inputs: List, input tensors for model
+            if not None, then the weights are built for both the source and self.
+        """
+        if inputs is not None:
+            # Ensure weights for both models are built.
+            if source_model:
+                source_model.model(inputs)
+            self.model(inputs)
+
+        self._ensure_built()
+
+        if source_model is not None:
+            for name, module in source_model.components.items():
+                if components is None or name in components:
+                    self.components[name] = module
+            self.build_model()
+
+        elif source_model is None:
+            self.restore(components=components,
+                         checkpoint=checkpoint,
+                         model_dir=model_dir)
+
+    def save_checkpoint(self, max_checkpoints_to_keep=5, model_dir=None):
```

Comment:
Let's add a docstring here that explains the idea of what this code is doing. It's a little confusing on first read of what's happening